prow/cmd/deck/static/prow/prow.ts

```diff
@@ -686,7 +686,11 @@ function createRerunCell(modal: HTMLElement, rerunElement: HTMLElement, prowjob:
                         method: 'post',
                     });
                     const data = await result.text();
-                    rerunElement.innerHTML = data;
+                    if (data === "needs_login") {
```

Comment:
If you switch to using a different status code (e.g. 401) as I mentioned in `prow/cmd/deck/main.go`, you could use something like:
> 
```suggestion
                    if (result.status === 401) {
```